### PR TITLE
read public keys

### PR DESCRIPTION
**Pull request by calvinmetcalf**
my c++ isn't very good so I may have made an error but the tests pass.

### CODE REVIEW

File: test/api-test.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 var assert = require('assert');
 var crypto = require('crypto');
 var ecdsa = require('../');
-
-var key = require('fs').readFileSync(__dirname + '/keys/key.pem');
-
+var fs = require('fs');
+var key = fs.readFileSync(__dirname + '/keys/key.pem');
+var pubkey = fs.readFileSync(__dirname + '/keys/pub.pem');
 describe('raw-ecdsa', function() {
```
Comment (indutny) [inline test/api-test.js:8]: Newlines? Where are they? :)

File: test/api-test.js [hunk 2]
```diff
@@ -23,7 +23,15 @@ describe('raw-ecdsa', function() {
     assert(!priv.verify(s, hash2), 'verify');
   });
 });
-
+describe('public key', function() {
```
Comment (indutny) [inline test/api-test.js:31]: Newlines

File: test/api-test.js [hunk 3]
```diff
@@ -23,7 +23,15 @@ describe('raw-ecdsa', function() {
...
+  it('should sign/verify data', function() {
+    var priv = new ecdsa.Key(key);
+    var pub = new ecdsa.Key(pubkey);
+    var hash = crypto.createHash('sha1').update('hello world').digest();
+    var s = priv.sign(hash);
+    assert(pub.verify(s, hash), 'verify');
+  });
+});
 describe('raw-ecdsa', function() {
```
Comment (indutny) [inline test/api-test.js:41]: Newlines

File: src/rawecdsa.cc [hunk 4]
```diff
@@ -59,7 +62,6 @@ class Key : public Nan::ObjectWrap {
 
     EC_KEY* ec;
     EVP_PKEY* evp = NULL;
-
```
Comment (indutny) [inline src/rawecdsa.cc:62]: Unnecessary change.

File: src/rawecdsa.cc [hunk 5]
```diff
@@ -8,7 +8,10 @@
 #include "openssl/pem.h"
 #include "openssl/x509.h"
 #include "v8.h"
+#include <string.h>
 
+static const unsigned char PUBLIC_KEY_PFX[] =  "-----BEGIN PUBLIC KEY-----";
+static const int PUBLIC_KEY_PFX_LEN = sizeof(PUBLIC_KEY_PFX) - 1;
```
Comment (indutny) [inline src/rawecdsa.cc:15]: newline after this.

File: src/rawecdsa.cc [hunk 6]
```diff
@@ -8,7 +8,10 @@
 #include "openssl/pem.h"
 #include "openssl/x509.h"
 #include "v8.h"
+#include <string.h>
```
Comment (indutny) [inline src/rawecdsa.cc]: Move this to the very top of the file and separate it from the `#include "node.h"` by a newline.

File: src/rawecdsa.cc [hunk 7]
```diff
@@ -90,6 +95,8 @@ class Key : public Nan::ObjectWrap {
     Key* k = new Key(evp, ec);
     k->Wrap(info.This());
 
+
```
Comment (indutny) [inline src/rawecdsa.cc]: Why this changes?

Comment (calvinmetcalf) [inline src/rawecdsa.cc]: not intentional

File: test/api-test.js [hunk 8]
```diff
@@ -37,4 +56,5 @@ describe('raw-ecdsa', function() {
     var hash2 = crypto.createHash('sha256').update('hello world2').digest();
     assert(!priv.verify(s, hash2));
   });
+
```
Comment (indutny) [inline test/api-test.js]: No need in this :)

File: test/api-test.js [hunk 9]
```diff
@@ -1,30 +1,49 @@
...
+    var pub = new ecdsa.Key(pubkey);
+    var hash = crypto.createHash('sha1').update('hello world').digest();
+    var s = priv.sign(hash);
+    assert(pub.verify(s, hash), 'verify');
+  });
+
 });
 
 describe('raw-ecdsa', function() {
+
```
Comment (indutny) [inline test/api-test.js]: Ditto.

File: test/api-test.js [hunk 10]
```diff
@@ -1,30 +1,49 @@
...
     var hash = crypto.createHash('sha1').update('hello world2').digest();
     var hash2 = crypto.createHash('sha224').update('hello world2').digest();
     var s = priv.sign(hash);
     assert(!priv.verify(s, hash2), 'verify');
   });
+  
+});
+
+describe('public key', function() {
+
```
Comment (indutny) [inline test/api-test.js]: Ditto.

File: test/api-test.js [hunk 11]
```diff
@@ -1,30 +1,49 @@
...
+  it('should sign/verify data', function() {
+    var priv = new ecdsa.Key(key);
+    var pub = new ecdsa.Key(pubkey);
+    var hash = crypto.createHash('sha1').update('hello world').digest();
+    var s = priv.sign(hash);
+    assert(pub.verify(s, hash), 'verify');
+  });
+
```
Comment (indutny) [inline test/api-test.js]: Ditto.

File: test/api-test.js [hunk 12]
```diff
@@ -1,30 +1,49 @@
...
     assert(priv.verify(s2, hash2), 'verify 2');
   });
+
   it('should not sign/verify bad data', function () {
     var hash = crypto.createHash('sha1').update('hello world2').digest();
     var hash2 = crypto.createHash('sha224').update('hello world2').digest();
     var s = priv.sign(hash);
     assert(!priv.verify(s, hash2), 'verify');
   });
+
```
Comment (indutny) [inline test/api-test.js]: Ditto.

File: test/api-test.js [hunk 13]
```diff
@@ -1,30 +1,49 @@
...
 var crypto = require('crypto');
 var ecdsa = require('../');
-
-var key = require('fs').readFileSync(__dirname + '/keys/key.pem');
+var fs = require('fs');
+var key = fs.readFileSync(__dirname + '/keys/key.pem');
+var pubkey = fs.readFileSync(__dirname + '/keys/pub.pem');
 
 describe('raw-ecdsa', function() {
+
```
Comment (indutny) [inline test/api-test.js]: Ditto.

**Comment by indutny** [comment]
Some minor nits, will land once they will be fixed. Thank you!

**Comment by calvinmetcalf** [comment]
woops went overboard on the newlines :)

**Comment by indutny** [comment]
Fantastic! Let's squash it down and give it an appropriate commit message: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-3-commit

**Comment by calvinmetcalf** [comment]
squashed

**Comment by indutny** [comment]
@calvinmetcalf let's do it like this `src: better support for public keys`?

**Comment by calvinmetcalf** [comment]
ok

**Comment by indutny** [comment]
Landed in 51ad05c, thank you!

**Comment by indutny** [comment]
Released in 1.1.0
